Source/Core/Common/OpenXR.h

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <openxr/openxr.h>
+
+#include <atomic>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "Common/MathUtil.h"
+#include "Common/Matrix.h"
+
+namespace OpenXR
+{
+bool IsInit();
+
+bool Init();
+bool Shutdown();
+
+void EventThreadFunc();
+
+class Session
+{
+  friend void EventThreadFunc();
+
+public:
+  Session() = default;
+  ~Session();
+
+  Session(const Session&) = delete;
+  Session& operator=(const Session&) = delete;
+
+  bool Create(const std::vector<std::string_view>& required_extensions,
+              const void* graphics_binding);
+
+  bool Destroy();
+
+  bool IsValid() const;
+
+  bool CreateSwapchain(const std::vector<int64_t>& supported_formats);
```

Comment:
`s64`, etc where applicable in the header.

Reply:
I was trying to be consistent with the types used in the OpenXR spec here which uses int64_t but I can change it to s64.